### PR TITLE
Task/modal integrantes

### PR DESCRIPTION
**Pull request by csssaz**
#### What's this PR do?

It changes the way in which we add and edit integrantes of a family.

#### Where should the reviewer start?

captura/views.py
familias/forms.py
familias/tests.py
captura/test_views.py

check the static files, including the js.
#### How should this be manually tested?
npm install
run the tests.
log as a capturista and try adding/editing integrantes.
#### Any background context you want to provide?
The main purpose of this PR is to allow the creation and modification of integrantes in modals, via ajax. Instead of three different forms for integrante, alumno and tutor, there's just one that inherits from the modelform of integrante.

This template was adapted ~stolen~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: captura/views.py [hunk 1]
```diff
@@ -345,199 +344,66 @@ def integrantes(request, id_familia):
...
-    return render(request, 'captura/captura_base.html', context)
+    if request.is_ajax() and request.method == 'GET':
+        integrante = Integrante.objects.get(pk=id_integrante)
+        initial_data = {}
+        rol = IntegranteForm.OPCION_ROL_NINGUNO
+        if hasattr(integrante, 'alumno_integrante'):  # check reverse relation w/alumno
+            rol = IntegranteForm.OPCION_ROL_ALUMNO
+            initial_data['numero_sae'] = integrante.alumno_integrante.numero_sae
+            initial_data['escuela'] = integrante.alumno_integrante.escuela
+        elif hasattr(integrante, 'tutor_integrante'):  # check reverse relation w/tutor
```
Comment (fernandolobato) [inline captura/views.py]: a lot more elegante than try catch

File: familias/forms.py [hunk 2]
```diff
@@ -54,43 +56,78 @@ def __init__(self, *args, **kwargs):
...
+    OPCIONES_RELACION = (('', '---------'),) + Tutor.OPCIONES_RELACION
+    escuela = ModelChoiceField(required=False, queryset=Escuela.objects.all())
+    numero_sae = CharField(required=False, max_length=30)
+    relacion = ChoiceField(required=False, choices=OPCIONES_RELACION)
 
     def __init__(self, *args, **kwargs):
         # This adds the class form control to every single input field.
         # Implemented for bootstrap purposes.
-        super(AlumnoForm, self).__init__(*args, **kwargs)
+        super(IntegranteModelForm, self).__init__(*args, **kwargs)
```
Comment (fernandolobato) [inline familias/forms.py]: Am probably wrong but, is this necessary? Calling IntegranteModelForm super from a IntegranteModelForm is needed?

File: familias/forms.py [hunk 3]
```diff
@@ -54,43 +56,78 @@ def __init__(self, *args, **kwargs):
...
+        has the necessary fields in the cleaned_data.
+        """
+        cleaned_data = super(IntegranteModelForm, self).clean()
+        if cleaned_data['rol'] == IntegranteForm.OPCION_ROL_ALUMNO:
+            if not cleaned_data['numero_sae'] or not cleaned_data['escuela']:
+                raise ValidationError('El estudiante necesita el número sae y la escuela')
+            if cleaned_data['relacion']:
+                raise ValidationError('El estudiante no tiene relación')
+            return cleaned_data
+        if cleaned_data['rol'] == IntegranteForm.OPCION_ROL_TUTOR:
```
Comment (fernandolobato) [inline familias/forms.py]: A couple of break lines would be appreciated for legibility  🇼🇸

File: jp2_online/static/js/captura/captura_integrantes.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
+$(document).ready(function() {
+  $('#tablaCapturista').dataTable();
+  $('#id_fecha_de_nacimiento').datepicker({
+    changeMonth: true,
+    changeYear: true}
+  );
+  // hide inputs from creation form
+  $('#modal_create_integrante #_id_escuela').hide();
+  $('#modal_create_integrante #_id_numero_sae').hide();
+  $('#modal_create_integrante #_id_relacion').hide();
```
Comment (fernandolobato) [inline jp2_online/static/js/captura/captura_integrantes.js]: Not necessary for code review. But you can call only the id's with "_" prefix #_id_escuela, even if they are inside the modal they should be unique in the page.

File: familias/forms.py (hunk 2 above)
Comment (csssaz) [inline familias/forms.py]: im not 100% sure but i'm doing it so the things in line 50 are executed, and the __init__ of the forms.ModelForm is executed too.

**Comment by AndreeToledo** [comment]
👍

**Pull request merged**